tests/ext/test_application.py

```diff
@@ -1432,15 +1432,22 @@ async def callback(update, context):
     )
     def test_run_polling_basic(self, app, monkeypatch, caplog):
         exception_event = threading.Event()
+        exception_testing_done = threading.Event()
         update_event = threading.Event()
         exception = TelegramError("This is a test error")
         assertions = {}
 
         async def get_updates(*args, **kwargs):
             if exception_event.is_set():
                 raise exception
+
             # This makes sure that other coroutines have a chance of running as well
-            await asyncio.sleep(0)
+            if exception_testing_done.is_set() and app.updater.running:
+                # the longer sleep makes sure that we can exit also while get_updates is running
+                await asyncio.sleep(20)
```

Comment:
does it really have to be 20 secs? Doesn't that mean the test suite is hung up by 20 secs?

Reply:
no, what I'm testing here is that calling `Updater.stop` does indeed *not* wait for the 2 seconds to finish, i.e. that  `asyncio.wait` returns the `event.wait()` as done :)